persistence/sql/migrations/sql/20230104193739000000_courier_list_index.up.sql

```diff
@@ -0,0 +1,5 @@
+CREATE INDEX courier_messages_nid_created_at_id_idx ON courier_messages (nid, created_at DESC, id);
+
+CREATE INDEX courier_messages_nid_status_created_at_id_idx ON courier_messages (nid, status, created_at DESC, id);
+
+CREATE INDEX courier_messages_nid_recipient_created_at_id_idx ON courier_messages (nid, recipient, created_at DESC, id);
```

Comment:
Is `id` on this index neccessary? Do we also need an index for `nid, status, recipient`? By the waay, it would be more efficient to change the index order to `nid, recipient, status`. That would require us to change

```
	q := p.GetConnection(ctx).Where("nid=?", p.NetworkID(ctx))
	if filter.Status != nil {
		q = q.Where("status=?", *filter.Status)
	}
	if filter.Recipient != "" {
 		q = q.Where("recipient=?", filter.Recipient)
 	}
```

to


```
	q := p.GetConnection(ctx).Where("nid=?", p.NetworkID(ctx))
	if filter.Recipient != "" {
 		q = q.Where("recipient=?", filter.Recipient)
 	}
	if filter.Status != nil {
		q = q.Where("status=?", *filter.Status)
	}
```

Recipient has a higher cardinality than status, making it a better candidate for the index